File: clients/java/examples/collections-update-one-vectorize.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.collections.Collection;
import com.datastax.astra.client.collections.commands.Update;
import com.datastax.astra.client.collections.commands.options.CollectionUpdateOneOptions;
import com.datastax.astra.client.collections.commands.results.CollectionUpdateResult;
import com.datastax.astra.client.collections.definition.documents.Document;
import com.datastax.astra.client.core.query.Sort;

public class Example {

  public static void main(String[] args) {
    // Get an existing collection
    Collection<Document> collection =
        new DataAPIClient("**APPLICATION_TOKEN**")
            .getDatabase("**API_ENDPOINT**")
            .getCollection("**COLLECTION_NAME**");

    // Update a document
    CollectionUpdateOneOptions options =
        new CollectionUpdateOneOptions().sort(Sort.vectorize("Text to vectorize"));

    Update update = Update.create().set("color", "blue");

    CollectionUpdateResult result = collection.updateOne(null, update, options);

    System.out.println(result.getMatchedCount());
    System.out.println(result.getModifiedCount());
  }
}

File: clients/go/examples/collections-update-one-vectorize.go
package main

import (
	"context"
	"fmt"
	"log"

	"github.com/datastax/astra-db-go/v2/astra"
	"github.com/datastax/astra-db-go/v2/astra/options"
	"github.com/datastax/astra-db-go/v2/astra/sort"
	"github.com/datastax/astra-db-go/v2/astra/update"
)

func main() {
	ctx := context.Background()

	// Get an existing collection
	client := astra.NewClient()

	database := client.Database(
		"**API_ENDPOINT**",
		options.API().SetToken("**APPLICATION_TOKEN**"),
	)

	collection := database.Collection("**COLLECTION_NAME**")

	// Update a document
	result, err := collection.UpdateOne(
		ctx,
		nil,
		update.Coll().Set("color", "blue"),
		options.CollectionUpdateOne().
			SetSort(sort.Vectorize("Text to vectorize")),
	)
	if err != nil {
		log.Fatal(err)
	}

	fmt.Println(result)
}
